### PR TITLE
Improve initializers

### PR DESCRIPTION
**Pull request by unijewski**
- move all initializers to proper directory
- improve loading roda plugins
- set root path for the app

### CODE REVIEW

File: config/initializers/auto_loader.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Dir[Funguru.root + '/config/initializers/*.rb'].each do |file|
```
Comment (tomajask) [inline config/initializers/auto_loader.rb]: one liners ?

File: app.rb [hunk 2]
```diff
@@ -5,16 +5,23 @@
...
-  Dir['./validation_schemas/**/*.rb'].each { |f| require f }
+  opts[:root] = File.dirname(__FILE__)
+
+  class << self
+    def root
+      @root ||= opts[:root]
+    end
+  end
+
+  PLUGINS = %i(
```
Comment (tomajask) [inline app.rb]: Would it be available as global constant?

File: config/initializers/auto_loader.rb (hunk 1 above)
Comment (unijewski) [inline config/initializers/auto_loader.rb]: it might be

File: app.rb [hunk 3]
```diff
@@ -5,16 +5,23 @@
...
+    all_verbs indifferent_params json multi_route sinatra_helpers
+  ).freeze
+
+  # Load Roda plugins
+  PLUGINS.each { |plugin_name| plugin plugin_name }
+
+  # Load all initializers
+  require root + '/config/initializers/auto_loader.rb'
```
Comment (tomajask) [inline app.rb]:
```ruby
File.join(root, '/config/initializers/auto_loader.rb')
```
Might be a little bit nicer :)

File: config/initializers/auto_loader.rb (hunk 1 above)
Comment (tomajask) [inline config/initializers/auto_loader.rb]: Also here `File.join` might be a bit nicer

File: app.rb (hunk 2 above)
Comment (unijewski) [inline app.rb]: will change to be private

File: config/initializers/auto_loader.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+PATHS = %w(
```
Comment (tomajask) [inline config/initializers/auto_loader.rb]: This one will be available as global constant. Check it, please :) I think, we don't need it to be global.
Just a simple variable should be enough in this case.

File: config/initializers/auto_loader.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  config/initializers/*.rb
+  models/**/*.rb
+  routes/**/*.rb
+  services/**/*.rb
+  validation_schemas/**/*.rb
+).freeze
```
Comment (tomajask) [inline config/initializers/auto_loader.rb]:
```ruby
).map(&:freeze).freeze
```
Let's freeze also strings inside already frozen array :)

**Pull request merged**